nixos/lib/test-driver/test-driver.py

```diff
@@ -965,6 +951,28 @@ def serial_stdout_off() -> None:
     log._print_serial_logs = False
 
 
+class EnvDefault(argparse.Action):
+    """An argpars Action that takes values from the specified
+    environment variable as the flags default value.
+    """
+
+    def __init__(self, envvar, required=False, default=None, nargs=None, **kwargs):  # type: ignore
+        if not default and envvar:
+            if envvar in os.environ:
+                if nargs is not None and (nargs.isdigit() or nargs in ["*", "+"]):
+                    default = os.environ[envvar].split()
+                else:
+                    default = os.environ[envvar]
+        if required and default:
+            required = False
+        super(EnvDefault, self).__init__(
+            default=default, required=required, nargs=nargs, **kwargs
```

Comment:
It would be nice to add the default value if it exists. When executing `./result/bin/nixos-test-driver --help`, it would make explicit some values are already defined.
Otherwise, the script behavior is difficult to understand.

Reply:
unfortunately putting a newline there seems not supported by `argparse` without further ado.

```console
vlans="1 2" startScripts="nixos/lib/test-driver/test-driver.py nixos/lib/test-driver/test-driver.py nixos/lib/test-driver/test-driver.py" nixos/lib/test-driver/test-driver.py --help
usage: nixos-test-driver [-h] [-K] [-I] [--start-scripts [START-SCRIPT [START-SCRIPT ...]]] [--vlans [VLAN [VLAN ...]]] testscript

positional arguments:
  testscript            the test script to run

optional arguments:
  -h, --help            show this help message and exit
  -K, --keep-vm-state   re-use a VM state coming from a previous run
  -I, --interactive     drop into a python repl and run the tests interactively
  --start-scripts [START-SCRIPT [START-SCRIPT ...]]
                        start scripts for participating virtual machines (default from environment: ['nixos/lib/test-driver/test-driver.py', 'nixos/lib/test-driver/test-driver.py', 'nixos/lib/test-driver/test-driver.py'])
  --vlans [VLAN [VLAN ...]]
                        vlans to span by the driver (default from environment: ['1', '2'])
```

Reply:
Note the test driver doesn't need anymore because of pep8:
```
-                kwargs["help"] = kwargs["help"] + f" (default from environment: {default})"
+                kwargs["help"] = (
+                    kwargs["help"] + f" (default from environment: {default})"
+                )
```

Would it be possible to also print the `testscript` default value?

I agree it would be better to print the default on a new line, but everything on one line is acceptable i think.

Reply:
> Would it be possible to also print the testscript default value?


```console
positional arguments:
  testscript            the test script to run (default from environment: path/to/test/driver)
```

Reply:
I fixed the [format violation](https://github.com/NixOS/nixpkgs/compare/25a4282e3106f11979907a6a7bf23a033fdfea4c..fdd8b3df915b1e5cfeaf82ef69f25e586fbf1408).

Reply:
Yeah, help message is really nice!